test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -0,0 +1,533 @@
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>",
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option value=\"1\">One</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>",
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>"
+                    },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(DropDownListDataSet))]
+        public void DropDownList_WithNullModel_GeneratesExpectedValue(
+            IEnumerable<SelectListItem> selectList,
+            string expectedHtml,
+            string ignoredHtml)
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+            var savedDisabled = selectList.ToDictionary(item => item.Text, item => item.Disabled);
+            var savedGroup = selectList.ToDictionary(item => item.Text, item => item.Group);
+            var savedSelected = selectList.ToDictionary(item => item.Text, item => item.Selected);
+            var savedValue = selectList.ToDictionary(item => item.Text, item => item.Value);
+
+            // Act
+            var html = helper.DropDownList("Property1", selectList, optionLabel: null, htmlAttributes: null);
+
+            // Assert
+            Assert.Equal(expectedHtml, html.ToString());
+            Assert.Equal(savedDisabled, selectList.ToDictionary(item => item.Text, item => item.Disabled));
```

Comment:
I think Xunit's simply treating dictionaries as IEnumerables and doing the equivalent of SequenceEqual. Wouldn't it be simpler to Assert the values and ignore the key entirely?

`var savedDisabled = selectList.Select(item => item.Disabled).ToList();`
Assert.Equal(savedDisabled, selectList.Select(item => item.Disabled));